internal/alb/lb/attributes.go

```diff
@@ -0,0 +1,204 @@
+package lb
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/aws/aws-sdk-go/service/elbv2/elbv2iface"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/albctx"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/pkg/util/log"
+	api "k8s.io/api/core/v1"
+)
+
+const (
+	DeletionProtectionEnabledString = "deletion_protection.enabled"
+	AccessLogsS3EnabledString       = "access_logs.s3.enabled"
+	AccessLogsS3BucketString        = "access_logs.s3.bucket"
+	AccessLogsS3PrefixString        = "access_logs.s3.prefix"
+	IdleTimeoutTimeoutSecondsString = "idle_timeout.timeout_seconds"
+	RoutingHTTP2EnabledString       = "routing.http2.enabled"
+
+	DeletionProtectionEnabled = false
+	AccessLogsS3Enabled       = false
+	AccessLogsS3Bucket        = ""
+	AccessLogsS3Prefix        = ""
+	IdleTimeoutTimeoutSeconds = 60
+	RoutingHTTP2Enabled       = true
+)
+
+// Attributes represents the desired state of attributes for a load balancer.
+type Attributes struct {
+	// LbArn is the ARN of the load balancer
+	LbArn string
+
+	// DeletionProtectionEnabled: deletion_protection.enabled - Indicates whether deletion protection
+	// is enabled. The value is true or false. The default is false.
+	DeletionProtectionEnabled bool
+
+	// AccessLogsS3Enabled: access_logs.s3.enabled - Indicates whether access logs are enabled.
+	// The value is true or false. The default is false.
+	AccessLogsS3Enabled bool
+
+	// AccessLogsS3Bucket: access_logs.s3.bucket - The name of the S3 bucket for the access logs.
+	// This attribute is required if access logs are enabled. The bucket must
+	// exist in the same region as the load balancer and have a bucket policy
+	// that grants Elastic Load Balancing permissions to write to the bucket.
+	AccessLogsS3Bucket string
+
+	// AccessLogsS3Prefix: access_logs.s3.prefix - The prefix for the location in the S3 bucket
+	// for the access logs.
+	AccessLogsS3Prefix string
+
+	// IdleTimeoutTimeoutSeconds: idle_timeout.timeout_seconds - The idle timeout value, in seconds. The
+	// valid range is 1-4000 seconds. The default is 60 seconds.
+	IdleTimeoutTimeoutSeconds int64
+
+	// RoutingHTTP2Enabled: routing.http2.enabled - Indicates whether HTTP/2 is enabled. The value
+	// is true or false. The default is true.
+	RoutingHTTP2Enabled bool
+}
+
+func NewAttributes(attrs []*elbv2.LoadBalancerAttribute) (a *Attributes, err error) {
+	a = &Attributes{
+		DeletionProtectionEnabled: DeletionProtectionEnabled,
+		AccessLogsS3Enabled:       AccessLogsS3Enabled,
+		AccessLogsS3Bucket:        AccessLogsS3Bucket,
+		AccessLogsS3Prefix:        AccessLogsS3Prefix,
+		IdleTimeoutTimeoutSeconds: IdleTimeoutTimeoutSeconds,
+		RoutingHTTP2Enabled:       RoutingHTTP2Enabled,
+	}
+	for _, attr := range attrs {
+		switch aws.StringValue(attr.Key) {
```

Comment:
could be switch attrKey := aws.StringValue(attr.Key); attrKey{...

so avoids lot's of aws.StringValue(attr.Key) in the error report...
nitpicking too 😸